crates/red_knot_python_semantic/src/types/context.rs

```diff
@@ -73,6 +73,17 @@ impl<'db> InferContext<'db> {
         lint: &'static LintMetadata,
         node: AnyNodeRef,
         message: fmt::Arguments,
+    ) {
+        self.report_lint_with_secondary_messages(lint, node, message, vec![]);
+    }
+
+    /// Reports a lint located at `node`.
+    pub(super) fn report_lint_with_secondary_messages(
+        &self,
+        lint: &'static LintMetadata,
+        node: AnyNodeRef,
+        message: fmt::Arguments,
+        secondary_messages: Vec<SecondaryDiagnosticMessage>,
```

Comment:
I think you called this out too. We probably want something that scales a little better than a separate function for every combination of arguments :) But I'm fine with this for now